lib/tasks/create_most_ingredients.rake

```diff
@@ -0,0 +1,121 @@
+    token: ENV['GOOGLE_TOKEN_2']
+  )
+  meal = user.meals.create!(title: "Everything You Have", gut_feeling: 0)
+  dish = meal.dishes.create!(name: "All the bacon and eggs you have")
+
+  def create_foods_and_ingredients(foods, meal, dish)
+    foods.each do |food_data|
+      if food = Food.find_by(upc: food_data["gtinUpc"])
+        food.ingredients.each do |ingredient|
+          meal.meal_ingredients.create(ingredient: ingredient)
+        end
+      else
+        processed_food_data = process_food_data(food_data)
+
+          food = Food.create(
+            name: processed_food_data[:name],
+            brand: processed_food_data[:brand],
+            upc: processed_food_data[:upc]
+          )
+
+          processed_food_data[:ingredients].each do |name|
+            unless ingredient = Ingredient.find_by(name: name)
+              ingredient = Ingredient.create(name: name)
+            end
+            food.food_ingredients.create(ingredient: ingredient)
+            meal.meal_ingredients.create(ingredient: ingredient)
+          end
+          food.dish_foods.create(dish: dish)
+      end
+    end
+  end
+
+  def process_food_data(food_data)
+    if food_data['ingredients']
+      food = { name: food_data['description'],
+              brand: food_data['brandOwner'],
+                upc: food_data['gtinUpc'],
+        ingredients: process_ingredients(food_data['ingredients']) }
+    food
+    end
+  end
+
+  def process_ingredients(ingredient_list)
+    formatted_ingredients    = format_ingredients(ingredient_list)
+    consolidated_ingredients = consolidate_ingredients(formatted_ingredients).uniq.compact
+    validate_ingredients(consolidated_ingredients)
+  end
+
+  def consolidate_ingredients(ingredients)
+    ingredients.map! do |ingredient|
+      if ingredient.include?("MILK")
+        ingredient.replace("MILK") unless not_dairy_milk?(ingredient)
+      elsif ingredient.include?("SALT")
+        ingredient.replace("SALT")
+      else
+        ingredient
+      end
+    end
+  end
+
+  def not_dairy_milk?(ingredient)
+    ingredient.include?("ALMOND")  ||
+    ingredient.include?("SOY")     ||
+    ingredient.include?("CASHEW")  ||
+    ingredient.include?("COCONUT") ||
+    ingredient.include?("FLAX")    ||
+    ingredient.include?("HEMP")    ||
+    ingredient.include?("RICE")    ||
+    ingredient.include?("OAT")
+  end
+
+  def validate_ingredients(ingredients)
+      ingredients.delete_if { |ingredient| ingredient.include?("WATER") }
+      ingredients.delete_if { |ingredient| ingredient.include?("NATURAL FLAVOR") }
+      ingredients.delete_if { |ingredient| ingredient.include?("ARTIFICIAL FLAVOR") }
+  end
+
+  def format_ingredients(ingredient_list)
+    ingredient_list
+      .gsub(/ \[.*?\]/, '')
+      .gsub(/ \(.*?\)/, '')
+      .gsub(/\(\)\*/, '')
+      .gsub('ORGANIC ', '')
+      .split('.').first
+      .split(', 2%').first
+      .split(', CONTAINS').first
+      .split(', ')
+      .flatten
+  end
```

Comment:
Maybe pull these methods into a module? That would DRY up this file and the seeds file